src/os/bluestore/bluestore_tool.cc

```diff
@@ -219,6 +227,50 @@ void inferring_bluefs_devices(vector<string>& devs, std::string& path)
   }
 }
 
+void bluefs_import(
+  string input_file,
+  string dest_file,
+  CephContext *cct,
+  const string& path,
+  const vector<string>& devs)
+{
+  int r;
+  std::ifstream f(input_file.c_str(), std::ifstream::binary);
+  if (!f) {
+    r = -errno;
+    cerr << "open " << input_file.c_str() << " failed: " << cpp_strerror(r) << std::endl;
+    exit(EXIT_FAILURE);
+  }
+
+  std::unique_ptr<BlueFS> bs = std::unique_ptr<BlueFS>(open_bluefs(cct, path, devs));
+
+  BlueFS::FileWriter *h;
+  fs::path file_path(dest_file);
+  const string dir = file_path.parent_path();
+  const string file_name = file_path.filename();
+  r = bs->open_for_write(dir, file_name, &h, false);
+  if (r < 0) {
+    cerr << "Open " << dir << "/" << file_name << " failed: " << cpp_strerror(r) << std::endl;
+    exit(EXIT_FAILURE);
+  }
+  uint64_t max_block = 4096;
+  char buf[max_block];
+  uint64_t fsize = fs::file_size(input_file.c_str());
+  uint64_t left = fsize;
+  uint64_t size = 0;
+  while (left) {
+    size = std::min(max_block, left);
+    f.read(buf, size);
+    h->append(buf, size);
+    left -= size;
+  }
+  f.close();
```

Comment:
is this necessary?

Reply:
Is this "f.close()"?

Reply:
@zhangdaolong yes, i think `f` should close itself in the destructor.